### DISCUSSION TITLE
Backup - Velero or something else

### DISCUSSION THREAD

**Discussion by clickersmudge** [mysticaltech/terraform-hcloud-kube-hetzner#348] [Q&A] [answered]
Hi,
We have installed "Velero plugins for AWS" (https://github.com/vmware-tanzu/velero-plugin-for-aws) on the production cluster. We set up buckets on Amazon S3 and created a backup. On Hetzner, we created a new project to test the disaster recovery scenario. We set up a new cluster based on the "kube.tf" file from production.
Velero does a good job of restoring to a clean cluster. However, the cluster created with terraform already has services and Velero is not working properly. Does not return the cluster and projects to production state. Is there any way to install a minimal cluster setup? Can you recommend any proven way to backup the entire cluster, not only services?

**Comment by mysticaltech** [comment]
Hello @clickersmudge, unfortunately, I do not have personal experience with it. Just yesterday @JustinGuese was experimenting with Velero, maybe he can help.

I would bet that it's just a matter of just configuring Velero even more extensively, as our cluster is a standard one based on k3s, of course, the Terraform part is just the automation that launches things into orbit, but then it becomes a normal cluster.

**Reply by mysticaltech** [reply]
Please check this tutorial up https://picluster.ricsanfre.com/docs/backup/#velero-installation-and-configuration

**Comment by otavio** [comment]
I set it up here using longhorn for volume backup and velero for k8s and snapshot creation. Works fine.

**Comment by JustinGuese** [comment, marked as answer]
@clickersmudge yes I can confirm velero and restic working... We didn't have the longhorn setup, so I would say it's a longhorn thing?

Did you install velero via Helm or cli? For me the cli didn't work, but Helm worked out of the box.

Could you paste the logs of the velero and restic pod?

**Reply by aDingil** [reply]
I'm able to install it via cli but have problems that the restored pvcs arent "working" correctly.
Can you post your helm config for velero or did you use the link above posted by @mysticaltech

**Comment by codeagencybe** [comment]
Doesn't has Longhorn have similar feature build in for backups? Both volumes and cluster and DR backups (and restore)? 
Or am I missing out something important if I don't use Velero?

**Reply by mysticaltech** [reply]
It's for the full cluster, it can basically clone your cluster. Super useful for breaking version upgrades for instance.

**Reply by codeagencybe** [reply]
aha gotcha. So then basically Velero can be something additional on top of Longhorn backups? 
Or is it a replacement for Longhorn backup feature altogether? 
-> not replacing Longhorn entirely but only disable backup stuff in LH and use Velero instead?

**Reply by mysticaltech** [reply]
No no, on top!
